Guide.docc/DataRaceSafety.md

```diff
@@ -448,6 +448,53 @@ The Swift Programming Language.
 
 [Sendable Types]: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency#Sendable-Types
 
+#### Flow-Sensitive Isolation Analysis
+
+The `Sendable` protocol is used to express thread-safety for a type as a
+whole.
+But there are situations when a particular _instance_ of a non-`Sendable`
+type is being used in a safe way.
+The compiler is often capable of inferring this safety through
+flow-sensitive analysis known as [region-based isolation][RBI].
+
+Region-based isolation allows the compiler to permit instances of
+non-`Sendable` types to cross isolation domains when it can prove doing
+so cannot introduce data races.
+
+```swift
+func populate(island: Island) async {
+    let chicken = Chicken()
+
+    await island.adopt(chicken)
+}
+```
+
+Here, the compiler can correctly reason that even though `chicken` has a
+non-`Sendable` type, allowing it to cross into the `island` isolation domain is
+safe.
+However, this exception to `Sendable` checking is inherently contigent on
+the surrounding code.
+The compiler will still produce an error should any unsafe accesses to the
+`chicken` variable ever be introduced.
```

Comment:
Let's add an example here, it'll help understand what this does I think

```swift
 func populate(island: Island) async {
     let chicken = Chicken()

     await island.adopt(chicken)
    chicken.check() // oh no!
 }
 ```

Reply:
yeah you're right